event_profiling.c

```diff
@@ -0,0 +1,458 @@
+#include "event_profiling.h"
+
+/* Ruby */
+#include "ractor_core.h"
+#include "ruby.h"
+#include "ruby/atomic.h"
+#include "ruby/thread_native.h"
+#include "vm_core.h"
+
+#if USE_EVENT_PROFILING
+
+/* GCC warning */
+#include <sys/syscall.h>
+pid_t gettid(void) { return syscall(SYS_gettid); }
+
+/* Assertion */
+#define refute_greater_or_equal(var, compare, reason)                          \
+    do                                                                         \
+    {                                                                          \
+        if ((var) >= (compare))                                                \
+        {                                                                      \
+            fprintf(stderr, ("[ERROR] event_profiling: " reason), var);        \
+            exit(1);                                                           \
+        }                                                                      \
+    } while (0)
+
+/* Global config */
+event_profiling_config_t *rb_event_profiling_config;
+
+/* Global bucket */
+profiling_event_bucket_t *rb_profiling_event_bucket;
+
+/* Phase strings */
+static const char profiling_event_phase_str[] = {'B', 'E', 'O'};
+
+/* Internal functions */
+static inline int get_total_events()
```

Comment:
This is an old K&R style.

```suggestion
static inline int get_total_events(void)
```